tasmota/tasmota_xsns_sensor/xsns_44_sps30.ino

```diff
@@ -140,14 +140,17 @@ unsigned char cmdb[6];
 void SPS30_Detect(void)
 {
   if (!I2cSetDevice(SPS30_ADDR)) { return; }
-  I2cSetActiveFound(SPS30_ADDR, "SPS30");
-
   uint8_t dcode[32];
   sps30_get_data(SPS_CMD_GET_SERIAL,dcode,sizeof(dcode));
-  AddLog(LOG_LEVEL_DEBUG, PSTR("sps30 found with serial: %s"),dcode);
+  if(dcode[0] == 0) {
+    AddLog(LOG_LEVEL_DEBUG, PSTR("No serial returned, not an SPS30, maybe a SEN5X?"));
+    return;
+  }
+  AddLog(LOG_LEVEL_DEBUG, PSTR("sps30 found with serial: %s"), dcode);
   sps30_cmd(SPS_CMD_START_MEASUREMENT);
```

Comment:
I think there is a problem here that was already there before your PR so it may be an opportunity to fix it
The serial number is 32 ascii character but nothing say that there should be a '\0' anywhere
So the log with "%s" could endup very badly

I suggest to increase the dcode buffer to 33 and insure that ` dcode[32] is a '\0' before the log

Reply:
Yeah, I did have a good look, there's an initial nulling of the passed buffer by the get_data command, and as you said the datasheet says 32 including the trailing null.
I don't have an SPS30, but doing an equivalent test with placeholder values and no new assigned value worked as expected.